### ISSUE TITLE
Failed to access residuals after fitting FirstLevelModel

### ISSUE THREAD

**Issue by alexprz** [nistats/nistats#369]
Hi,

I'm trying to get the residuals of the FirstLevelModel once fitted on a 4D image using the resid() method of the RegressionResults class but I get this error : 
```
TypeError: resid() missing 1 required positional argument: 'Y'
```
That's unexpected since resid() does not require any argument according to the [documentation.
](https://nistats.github.io/modules/generated/nistats.regression.RegressionResults.html)

### Minimal reproductible example
```python
import nilearn
import pandas as pd
import numpy as np

from nistats.datasets import fetch_spm_auditory
from nistats.design_matrix import make_first_level_design_matrix
from nistats.first_level_model import FirstLevelModel

subject_data = fetch_spm_auditory()
fmri_img = nilearn.image.concat_imgs(subject_data.func, auto_resample=True)

n_scans = fmri_img.shape[-1]
events = pd.read_table(subject_data['events'])
frame_times = np.arange(n_scans) * 2.
design_matrix = make_first_level_design_matrix(frame_times, events)

fmri_glm = FirstLevelModel().fit(fmri_img, design_matrices=design_matrix)

for key, reg_results in fmri_glm.results_[0].items():
    reg_results.resid()
```

### Versions
```
nistats==0.0.1b0
nilearn==0.5.2
numpy==1.16.4
pandas==0.24.2
```

**Comment by jeromedockes** [comment]
try `FirstLevelModel(minimize_memory=False)`, that seems to fix your example (thanks for providing it BTW)

by default `FirstLevelModel` will store the results as `SimpleRegressionResults` (that in particular don't store the `Y` used during fit), but it will keep the full `RegressionResults`, whose documentation you point to, if you clear `minimize_memory`.

also, it is a property so use `reg_results.resid` (without the parentheses). the documentation is confusing and shows parentheses, I guess that sphinx gets confused because the property is not decorated with `@property` but some decorator from nibabel

**Comment by jeromedockes** [comment]
this line in the docstring

https://github.com/nistats/nistats/blob/9fa2e0e7cd8a0deed5f57d9645d080b64e671de2/nistats/first_level_model.py#L117

should be fixed to state that by defaults results are stored as `SimpleRegressionResults`, not `RegressionResults`

**Comment by alexprz** [comment]
Thanks this fixed it.  
It would be worth specifying it in the doc actually. We might take advantage of the opportunity to  match the FirstLevelModel attributes' names with the doc : 
https://github.com/nistats/nistats/blob/9fa2e0e7cd8a0deed5f57d9645d080b64e671de2/nistats/first_level_model.py#L316-L318
https://github.com/nistats/nistats/blob/9fa2e0e7cd8a0deed5f57d9645d080b64e671de2/nistats/first_level_model.py#L265-L272

**Comment by kchawla-pi** [comment]
@alexprz WOuld you be interested in submitting a PR to fix it?

**Comment by alexprz** [comment]
@kchawla-pi yes sure I'll take a look at it soon

**Comment by kchawla-pi** [comment]
Fixed in #371
